### PR TITLE
main: support generate declarative workflows

### PR DESCRIPTION
**Pull request by luantranminh**

### CODE REVIEW

File: .github/workflows/ci-go.yaml [hunk 1]
```diff
@@ -56,13 +56,13 @@ jobs:
       - name: Run GitHub atlas extension
         run: |
           gh atlas init-action -R ${{ secrets.TEST_REPO }} --token=${{ secrets.TEST_ATLAS_TOKEN }} --dir-name="sync-action-integration-test" --driver="sqlite" "sqlite_dir" \
-          --config-path="atlas.hcl" --config-env="local"
+          --config-path="atlas.hcl" --config-env="local" --flow=migrate
```
Comment (ronenlu) [inline .github/workflows/ci-go.yaml:59]: I think is should be subcommand, and not a flag.
i.e: ` gh atlas init-action migrate` and `gh atlas init-action schema`

Comment (luantranminh) [reply .github/workflows/ci-go.yaml:59]: Oh nice. Let me change this.

File: gen/gen.go [hunk 2]
```diff
@@ -4,15 +4,16 @@ import (
...
 	_ "embed"
-	"fmt"
-	"slices"
 	"text/template"
 )
 
 type (
 	Driver string
 	// Config passed to template parser
 	Config struct {
```
Comment (ronenlu) [inline gen/gen.go:13]: I think its better to store to structs in config, 1 for migrate, and 1 for schema:

```go
Config struct {
  Migrate *migrateConfig
  Schema *schemaConfig
  // global vars
  Env          string
  ...
}
```

Comment (luantranminh) [reply gen/gen.go:13]: @ronenlu i thought about it, but after trying, I didn’t see it bringing a huge advantage compared to using a single struct, and it increased code complexity.

**Comment by luantranminh** [comment]
waiting for nebula release

File: gen/declarative.tmpl [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          {{- with .ConfigPath }}
+          config: 'file://{{ . }}'
+          {{- end }}
+          {{- with .Env }}
+          env: '{{ . }}'
+          {{- end }}
+        env:
+          GITHUB_TOKEN: {{`${{ github.token }}`}}
+  apply:
+    if: {{`${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}`}}
```
Comment (giautm) [inline gen/declarative.tmpl:33]:
```suggestion
    if: {{`${{ github.event_name == 'push' && github.ref == 'refs/heads/{{ .DefaultBranch }}' }}`}}
```

File: prompt.go [hunk 4]
```diff
@@ -270,28 +411,22 @@ func (i *InitActionCmd) setConfigEnv(envs map[string]bool) error {
...
+		Templates: &promptui.PromptTemplates{
+			Success: fmt.Sprintf(`{{ "%s" | green }} {{ "Open in browser: " | faint }}`, promptui.IconGood),
+		},
 	}
-	if _, err := prompt.Run(); err != nil {
+	ans, err := prompt.Run()
+	if err != nil {
+		// https://github.com/manifoldco/promptui/issues/81
+		// the promptui library generates error if response is 'n'
+		if strings.ToLower(ans) == "n" {
```
Comment (giautm) [inline prompt.go:427]: Why don't compare with ErrAbort?
https://github.com/manifoldco/promptui/issues/81#issuecomment-495265273

File: gen/declarative.tmpl [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          {{- with .ConfigPath }}
+          config: 'file://{{ . }}'
+          {{- end }}
+          {{- with .Env }}
+          env: '{{ . }}'
+          {{- end }}
+        env:
+          GITHUB_TOKEN: {{`${{ github.token }}`}}
+      - uses: ariga/atlas-action/schema/plan/approve@master
+        id: schema-plan
```
Comment (giautm) [inline gen/declarative.tmpl:46]:
```suggestion
        id: plan-approve
```
Just for clear

File: gen/declarative.tmpl [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        env:
+          GITHUB_TOKEN: {{`${{ github.token }}`}}
+  apply:
+    if: {{`${{ github.event_name == 'push' && github.ref == 'refs/heads/`}}{{ .DefaultBranch }}{{`' }}`}}
+    {{- template "job-setup" . }}
+      - uses: ariga/atlas-action/schema/push@master
+        with:
+          {{- if .CreateDevURL }}
+          {{ template "UseServices" . }}
+          {{- end -}}
```
Comment (giautm) [inline gen/declarative.tmpl:36]: Template name should be lower case: `use-service`

But, I think this template should be a Go function that print URL depends on Driver.

Comment (luantranminh) [reply gen/declarative.tmpl:36]: I'll make it a func

**Comment by luantranminh** [comment]
Videos added for better review

**Review by giautm — APPROVED** [review]
LGTM, wait for @rotemtam and @a8m

File: prompt.go [hunk 7]
```diff
@@ -53,10 +119,104 @@ func (i *InitActionCmd) setDriver() error {
...
+		Stdin: i.stdin,
+		Validate: func(input string) error {
+			if len(i.From) == 0 && strings.TrimSpace(input) == "" {
+				return errors.New("at least one URL is required for current schema state")
+			}
+			return nil
+		},
+		Templates: &promptui.PromptTemplates{
+			Valid:   fmt.Sprintf(`{{ "%s" | green }} {{ . | faint }} {{": " | faint }}`, promptui.IconGood),
+			Success: fmt.Sprintf(`{{ "%s" | green }} {{ "Current schema state URL: " | faint }}`, promptui.IconGood),
```
Comment (a8m) [inline prompt.go:209]:
```suggestion
			Success: fmt.Sprintf(`{{ "%s" | green }} {{ "Current schema state URL:" | faint }}`, promptui.IconGood),
```
image

**Review by a8m — APPROVED** [review]

**Comment by luantranminh** [comment]
Thanks guys. 🚀🚀🚀🚀

**Pull request merged**